app/javascript/src/LoginPage/loginForms/Login3scaleForm.jsx

```diff
@@ -33,50 +39,40 @@ class Login3scaleForm extends React.Component<Props, State> {
     }
   }
 
-  setIsValidUsername = () => {
-    this.setState({isValidUsername: this.state.username !== ''})
-  }
-
-  handleTextInputUsername = (username: string) => {
-    this.setState({ username }, this.setIsValidUsername)
-  }
-
-  setIsValidPassword = () => {
-    this.setState({isValidPassword: this.state.password !== ''})
-  }
-
-  handleTextInputPassword = (password: string) => {
-    this.setState({ password }, this.setIsValidPassword)
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
+    const isValid = validateSingleField(event)
+    this.setState({
+      [namesToStateKeys[event.currentTarget.name].name]: value,
```

Comment:
Could we get rid of `nameToStateKeys`? It's a bit convoluted.. and it's not used everywhere like I think it should be, for instance in the Type definition the state keys are hardcoded and also in the constructor.

If not, why not use `value` instead of `event.currentTarget.name`?